### PR TITLE
[Merged by Bors] - fix: replaces custom llvm install script with action

### PR DESCRIPTION
**Pull request by EstebanBorai**
Uses `KyleMayes/install-llvm-action@v1` action to install LLVM when running
CI workflows.

Resolves: https://github.com/infinyon/fluvio/issues/2714

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -86,6 +86,10 @@ jobs:
         with:
           toolchain: ${{ matrix.rust }}
           profile: minimal
+      - name: Install LLVM and Clang
+        uses: KyleMayes/install-llvm-action@v1
+        with:
+          version: "14.0"
```
Comment (EstebanBorai) [inline .github/workflows/ci.yml:92]: Uses version 14 as we did for `FLUVIO_BUILD_LLD=lld-14`.

**Review by sehz — APPROVED** [review]
LGTM.  Nice job

**Pull request closed without merging**